src/services/user.service.ts

```diff
@@ -21,4 +22,65 @@ export class UserService implements IUserService {
         return await UserModel.insertMany([userRegisterDTO]);
     }
 
+    public async login(userLoginDTO: UserLoginDTO): Promise<UserLoginResponseDTO> {
+        const user = await UserModel.findOne({email: userLoginDTO.email});
+
+        if (!user?._id) {
+            throw new Unauthorized('wrong email or password');
+        }
+
+        if (!await compare(userLoginDTO.password, user?.password || '')) {
+            throw new Unauthorized('wrong email or password');
+        }
+
+        const token = await generateToken(user?._id, Date.now() / 1000 + config.exp)
+        const userLoginResponseDTO: UserLoginResponseDTO =  {
+            _id: user?._id,
+            token
+        }
+
+        await UserModel.updateOne(
```

Comment:
how can you be sure that update of user was correct?

Reply:
doesn't that line ensure that? 
"const user = await UserModel.findOne({email: userLoginDTO.email});"

anyways I updated the code.
what about insertMany function? should its result be checked as well?